### PR TITLE
Gradle: Minor improvements

### PR DESCRIPTION
**Pull request by bisgardo**
- Use "modern" style plugin declaration: We're using the [legacy](https://docs.gradle.org/current/userguide/plugins.html#sec:old_plugin_application) style which [should be avoided](https://stackoverflow.com/a/50153617/883073).
- Set the execution bit for `./gradlew`. It's pretty annoying that this script isn't checked in as executable, and it's hard to see a good reason for it to be that way. Gradle itself bootstraps new projects with the execution bit set.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
 
-apply plugin: 'com.android.application'
-apply plugin: 'kotlin-android'
-apply plugin: 'kotlin-kapt'
-apply plugin: "androidx.navigation.safeargs.kotlin"
+plugins {
+    id 'com.android.application'
+    id 'kotlin-android'
+    id 'kotlin-kapt'
+    id 'androidx.navigation.safeargs.kotlin'
```
Comment (bisgardo) [inline app/build.gradle:7]: Removing `androidx.navigation.safeargs.kotlin` doesn't break the build. Does this mean that we're not using this and might as well remove it? @sasho-axellero

Comment (sasho-axellero) [reply app/build.gradle:7]: It is required for the Navigation component below, but that is also not used.
The whole file should be cleaned up and libraries updated.
I don't know if this is something we should do with this update.

Comment (ali-concordium) [reply app/build.gradle:7]: In context of improving Gradle, I think it's fine "boyscouting" and removing dead code.
Timebox it to 1-2 hours please.

Comment (ali-concordium) [reply app/build.gradle:7]: If it's something minor, please go ahead.

**Review by sasho-axellero — APPROVED** [review]

File: app/build.gradle (hunk 1 above)
Comment (sasho-axellero) [reply app/build.gradle:7]: Ok, I'll clean and update

**Pull request merged**